src/test/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/SummaryReportServiceTest.java

```diff
@@ -0,0 +1,264 @@
+            objectMapper,
+            envelopeService,
+            serviceConfiguration
+        );
+    }
+
+    @Test
+    void should_continue_if_there_is_no_supplier_report() throws JsonProcessingException {
+        // given
+        LocalDate date = LocalDate.now();
+        given(supplierStatementRepository.findLatest(date)).willReturn(Optional.empty());
+
+        // when
+        summaryReportService.process(date);
+
+        // then
+        verifyNoInteractions(envelopeService);
+        verifyNoInteractions(reconciliationReportRepository);
+    }
+
+    @Test
+    void should_continue_process_with_next_target_storage_if_one_fails()
+        throws Exception {
+        // given
+        given(storageConfig.get(anyString())).willReturn(createStorageConfigItem(BULKSCAN));
+        LocalDate date = LocalDate.now();
+        String content = Resources.toString(
+            getResource("reconciliation/valid-supplier-statement.json"),
+            UTF_8
+        );
+
+        UUID supplierId = UUID.randomUUID();
+        var envelopeSupplierStatement = new EnvelopeSupplierStatement(
+            supplierId,
+            date,
+            content,
+            "1.0",
+            LocalDateTime.now()
+        );
+
+        given(supplierStatementRepository.findLatest(date))
+            .willReturn(Optional.of(envelopeSupplierStatement));
+        var envelopeList = Arrays.asList(
+            createEnvelope("1010404021234_14-08-2020-08-31.zip", "probate")
+        );
+        given(envelopeService.getEnvelopes(date)).willReturn(envelopeList);
+        given(reconciliationReportRepository.save(any()))
+            .willThrow(new RuntimeException("Can not save"));
+        //when
+        summaryReportService.process(date);
+
+        // then
+        verify(reconciliationReportRepository, times(3)).save(any());
+
+    }
+
+    @ParameterizedTest
+    @MethodSource("serviceWithClassification")
+    void should_save_reports_if_there_is_supplier_report(
+        String supplierStatementFile,
+        List<Envelope> envelopeList,
+        String summaryReportPostfix
+    ) throws IOException, SQLException, JSONException {
+        // given
+        setupStorageConfig();
+        LocalDate date = LocalDate.now();
+        String content = Resources.toString(
+            getResource(supplierStatementFile),
+            UTF_8
+        );
+
+        UUID supplierId = UUID.randomUUID();
+        var envelopeSupplierStatement = new EnvelopeSupplierStatement(
+            supplierId,
+            date,
+            content,
+            "1.0",
+            LocalDateTime.now()
+        );
+
+        given(supplierStatementRepository.findLatest(date)).willReturn(Optional.of(envelopeSupplierStatement));
+
+        given(envelopeService.getEnvelopes(date)).willReturn(envelopeList);
+
+        //when
+        summaryReportService.process(date);
+
+        //then
+        verify(reconciliationReportRepository, times(3))
```

Comment:
why 3?

Reply:
for all target storage it should try even 1 fails.

Reply:
`TargetStorageAccount.values().length`